File: java-api/src/main/java/com/webmvc/javaapi/config/security/UserDetailsServiceImpl.java
package com.webmvc.javaapi.config.security;

import com.webmvc.javaapi.repository.MemberRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

@RequiredArgsConstructor
public class UserDetailsServiceImpl implements UserDetailsService {

    private final MemberRepository memberRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        return memberRepository.findByAccount(username)
                .map(UserDetailsImpl::new)
                .orElseThrow(() -> new UsernameNotFoundException("유저 없음"));
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/global/config/UserDetailsServiceImpl.kt
package spring.web.kotlin.global.config

import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import spring.web.kotlin.domain.member.repository.MemberRepository

class UserDetailsServiceImpl(
    private val memberRepository: MemberRepository
) : UserDetailsService {
    override fun loadUserByUsername(username: String) =
        memberRepository.findByAccount(username).orElse(null)
            ?.let { UserDetailsImpl(it) }
            ?: throw UsernameNotFoundException("유저 없음")
}